### PR TITLE
Draft approach to enable Affirm and Afterpay payment methods

### PR DESCRIPTION
**Pull request by dmvrtx**
#### Changes proposed in this Pull Request

 - Investigation of steps required to add support for new payment methods

#### Testing instructions

* TBD: tests were not updated for this draft.

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5910` or branch name `add/draft-afterpay-affirm-integration` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/draft-afterpay-affirm-integration) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: dacb6756a35faf1380983f454021f1bc3c0d48fd
- Build time: 2023-04-03 14:20:04 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/payment-method-details/index.js [hunk 1]
```diff
@@ -41,6 +41,9 @@ const formatDetails = ( payment ) => {
 					{ paymentMethod.iban_last4 }
 				</Fragment>
 			);
+		case 'affirm':
```
Comment (kalessil) [inline client/components/payment-method-details/index.js:44]: Perhaps, it's opinionated, but I find the following construct easier to read:

```php
        case 'affirm':
		case 'afterpay_clearpay':
		default:
			return <Fragment />;
```

File: includes/payment-methods/class-affirm-payment-method.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	 *
+	 * @param WC_Payments_Token_Service $token_service Token class instance.
+	 */
+	public function __construct( $token_service ) {
+		parent::__construct( $token_service );
+		$this->stripe_id   = self::PAYMENT_METHOD_STRIPE_ID;
+		$this->title       = __( 'Affirm', 'woocommerce-payments' );
+		$this->is_reusable = false;
+		$this->currencies  = [ 'USD' ];
+		$this->icon_url    = plugins_url( 'assets/images/payment-methods/woo.svg', WCPAY_PLUGIN_FILE );
```
Comment (kalessil) [inline includes/payment-methods/class-affirm-payment-method.php]: `woo.svg`?

File: includes/payment-methods/class-afterpay-payment-method.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	 *
+	 * @param WC_Payments_Token_Service $token_service Token class instance.
+	 */
+	public function __construct( $token_service ) {
+		parent::__construct( $token_service );
+		$this->stripe_id   = self::PAYMENT_METHOD_STRIPE_ID;
+		$this->title       = __( 'Afterpay/Clearpay', 'woocommerce-payments' );
+		$this->is_reusable = false;
+		$this->currencies  = [ 'USD' ];
+		$this->icon_url    = plugins_url( 'assets/images/payment-methods/woo.svg', WCPAY_PLUGIN_FILE );
```
Comment (kalessil) [inline includes/payment-methods/class-afterpay-payment-method.php]: `woo.svg`?

Comment (dmvrtx) [reply includes/payment-methods/class-afterpay-payment-method.php]: Updated in 12a7d53d

File: includes/payment-methods/class-affirm-payment-method.php (hunk 2 above)
Comment (dmvrtx) [reply includes/payment-methods/class-affirm-payment-method.php]: Updated in 12a7d53d

File: client/components/payment-method-details/index.js (hunk 1 above)
Comment (dmvrtx) [reply client/components/payment-method-details/index.js:44]: Intention here was to indicate that an action is required before draft is ready for review.

Comment (kalessil) [reply client/components/payment-method-details/index.js:44]: Gotcha!

**Comment by kalessil** [comment]
For the next iteration:
- both methods are not reusable (hence can not be saved and used for purchasing subscriptions)
- also, referencing the gdoc, they have limitations on payment range which needs to be full-filled. Otherwise, PMs are not applicable
- a feature-flag so we could iteratively merge PRs until project is over

**Comment by KarlisJ** [comment]
@dmvrtx Could you describe the reasoning/problem for the "hacky" shipping data code?

**Comment by dmvrtx** [comment]
> @dmvrtx Could you describe the reasoning/problem for the "hacky" shipping data code?

@KarlisJ, Afterpay payment method requires shipping address to be sent along and it appears that Stripe.js version that we use doesn't fallback to the billing address if shipping is not present. Thus shipping is populated from billing in this "hacky" approach, but it should look if shipping address is provided first.

**Pull request closed without merging**